taskqueue/gcloud/aio/taskqueue/taskmanager.py

```diff
@@ -2,486 +2,159 @@
-        self.max_poll_interval = max_poll_interval
-        self.backoff = backoff(base=1.1, max_value=max_poll_interval)
-        self.max_concurrency = max_concurrency
+        self.backoff = backoff(base=backoff_base, factor=backoff_factor,
+                               max_value=backoff_max_value)
         self.batch_size = max(batch_size, 1)
+        self.deadletter_insert_function = deadletter_insert_function
+        self.lease_seconds = lease_seconds
         self.retry_limit = retry_limit
 
-        self.running = False
-        self.completed_tasks = 0
-        self.created_at = datetime.datetime.now()
-        self.failed = 0  # raw failures
-        self.flunked_tasks = 0  # failures that were sent to dead letter store
-        self.tasks_processing = {}
-        self.state = None
-        self.durations = RollingAverage(maxlen=100)
-        self.sizes = RollingAverage(maxlen=100)
-
-        call_later(1, self.print_stats)
-
-    @property
-    def no_tasks(self):
-
-        return len(self.tasks_processing.keys())
-
-    @property
-    def ids(self):
-
-        return sorted(self.tasks_processing.keys())
-
-    def start(self):
-
-        if self.running:
-            return self
+        self.semaphore = asyncio.BoundedSemaphore(max_concurrency)
+        self.tq = TaskQueue(project, service_file, taskqueue,
+                            location=location, session=session, token=token)
 
-        log.info('Starting task manager.')
-        self.running = True
-        fire(self.poll)
+        self.running = False
+        self.tasks = dict()
 
-        return self
+    async def autorenew(self, name):
+        while True:
+            await asyncio.sleep(self.lease_seconds / 2)
 
-    def stop(self):
+            task = await self.tq.renew(self.tasks[name],
+                                       lease_seconds=self.lease_seconds)
+            self.tasks[name] = task
 
-        log.info('Stopping task manager.')
-        self.running = False
+    async def fail(self, task, payload, exception):
+        if not self.deadletter_insert_function:
+            return
 
-    async def get_stats(self):
-
-        stats = await self.queue.get_stats()
-
-        return {
-            'task_queue': stats,
-            'instance': {
-                'tasks': {
-                    'completed': self.completed_tasks,
-                    'flunked': self.flunked_tasks,
-                    'failed': self.failed,
-                    'progressing': self.no_tasks,
-                    'rolling_duration': self.durations.avg,
-                    'rolling_size': self.sizes.avg
-                },
-                'created_at': str(self.created_at),
-                'state': getattr(self.state, 'name', None)
-            }
+        properties = {
+            'bucket': payload.get('bucket'),
+            'error': str(exception),
+            'generation': None,
+            'metageneration': None,
+            'time_created': datetime.datetime.now(),
```

Comment:
Do you have to worry about local time vs UTC conversions?

Reply:
Yes indeed. We weren't doing this before, but we may as well do it now.